### PR TITLE
Tidy up k8s ordering, directories and make bgp an experimental feature

### LINKED ISSUE #750: Need doc to support users upgrading from KDD on 2.1
Users who have installed according to this doc:
http://docs.projectcalico.org/v2.1/getting-started/kubernetes/installation/hosted/k8s-backend/

Need to continue to be supported when they upgrade to calico 2.2.  

The 2.2 KDD docs have manifests which enable BGP routing and don't mention the scenario from the 2.1 docs, leading to confusion about support.

Options:
- Continue to maintain a policy-only manifest, OR
- Document that the KDD manifest now enables BGP networking by default and include instructions for how to disable it for policy-only

### LINKED ISSUE #751: Choose label (beta?) for "KDD /w routing" and make it so for v2.2 release
Also, make a decision on whether we restore the KDD without routing manifest (discussed with @caseydavenport and we think we'll probably leave it as is but add a comment in the manifest describing how to turn off routing)

### PR DESCRIPTION
**Pull request by robbrockbank**
Fixes #750 

Fixes #751

### CODE REVIEW

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+              value: "ACCEPT"
+            # Disable IPV6 on Kubernetes.
+            - name: FELIX_IPV6SUPPORT
+              value: "false"
+            # Wait for the datastore.
+            - name: WAIT_FOR_DATASTORE
+              value: "true"
+            # The Calico IPv4 pool to use.  This should match `--cluster-cidr`
+            - name: CALICO_IPV4POOL_CIDR
+              value: "192.168.0.0/16"
```
Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml]: This pool needs to be switched also

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.5/calico.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+              value: "ACCEPT"
+            # Disable IPV6 on Kubernetes.
+            - name: FELIX_IPV6SUPPORT
+              value: "false"
+            # Wait for the datastore.
+            - name: WAIT_FOR_DATASTORE
+              value: "true"
+            # The Calico IPv4 pool to use.  This should match `--cluster-cidr`
+            - name: CALICO_IPV4POOL_CIDR
+              value: "192.168.0.0/16"
```
Comment (djosborne) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.5/calico.yaml]: Haven't given this much thought yet, but I think `CALICO_IPV4POOL_CIDR` is only applied if no pool exists. So the upgrade path could be fine here

Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.5/calico.yaml]: Yeah true dat. :slightly_smiling_face:

Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.5/calico.yaml]: Yup - no change required here I believe.

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml (hunk 1 above)
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml]: @tmjd :  what do you mean by switched?

Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml]: That was when I thought 192.168 should be 'switched' to 10.244. so disregard this comment

Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml]: I assume this is ok staying too?

Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/policy-only/1.6/calico.yaml]: Yep, it should be ok to stay as it is.

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md [hunk 3]
```diff
@@ -22,25 +25,80 @@ You must have a cluster which meets the following requirements:
...
 - Your Kubernetes controller manager has been provided a cluster-cidr (i.e. by passing `--cluster-cidr=192.168.0.0/16`, which the manifest expects by default).
 
+> Note:  If you are upgrading from Calico v2.1, the cluster-cidr selected for your controller manager should remain
+> unchanged from the v2.1 install (the v2.1 manifests default to `10.244.0.0/16`).
+
 ## Installation
 
-To install Calico, ensure you have a cluster which meets the above requirements and run one of the following commands based on your Kubernetes version:
+This document describes three installation options for the two operational modes
+for Calico (Calico with three installation options:
```
Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: Looks like a half edit.  You can probably just get rid of `(Calico with three installation options`

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md [hunk 4]
```diff
@@ -22,25 +25,80 @@ You must have a cluster which meets the following requirements:
...
+
+> Note:  There is currently no upgrade path to switch between different installation options.  Therefore, 
+> if you are upgrading from Calico v2.1, use the [Calico in policy-only mode](#calico-in-policy-only-mode) installation instructions 
+> to upgrade Calico policy-only leaving the networking solution unchanged.
+
+### Calico policy with Calico networking (Beta)
+
+With Kubernetes as the Calico datastore, Calico has Beta support for Calico networking.  This provides BGP-based
+networking with a full node-to-node mesh.  It is not currently possible to configure the Calico BGP network to peer with
+other routers - a future release will provide this feature, and to provide feature parity with the [Kubeadm Hosted Install](../kubeadm)
```
Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: This sentence doesn't flow well for me.  What about something like `It is not currently possible to configure the Calico BGP network to peer with other routers - future releases of Calico are expected to bring feature parity with the etcd backed Calico.`

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md [hunk 5]
```diff
@@ -22,25 +25,79 @@ You must have a cluster which meets the following requirements:
...
+kubectl apply -f {{site.url}}/{{page.version}}/getting-started/kubernetes/installation/hosted/kubernetes-datastore/calico-networking/1.5/calico.yaml
 ```
 
->[Click here to view the above yaml directly.](calico.yaml)
+>[Click here to view the above yaml directly.](calico-networking/1.5/calico.yaml)
+
+### Calico policy-only with user-supplied networking
+
+If you run Calico in policy-only mode it is necessary to configure your network to route pod traffic based on pod 
+CIDR allocations, either through static routes, a Kubernetes cloud-provider integration, or flannel.
```
Comment (tmjd) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: Did you purposefully include flannel here even though there is a dedicated section for it?

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md (hunk 3 above)
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: Yup - that fix is already pushed up...

File: master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md (hunk 4 above)
Comment (robbrockbank) [inline master/getting-started/kubernetes/installation/hosted/kubernetes-datastore/index.md]: yeah I agree - flow is not good.  will update.

**Pull request merged**